test/account-manager.js

```diff
@@ -0,0 +1,362 @@
+        })
+        let options = { multiUser, store, host }
+        let accountManager = AccountManager.from(options)
+
+        return accountManager.accountExists()
+          .then(exists => {
+            expect(exists).to.be.true
+          })
+      })
+
+      it('resolves to false if root .acl does not exist in root storage', () => {
+        let store = new LDP({
+          root: testAccountsDir,
+          idp: multiUser
+        })
+        let options = { multiUser, store, host }
+        let accountManager = AccountManager.from(options)
+
+        return accountManager.accountExists()
+          .then(exists => {
+            expect(exists).to.be.false
+          })
+      })
+    })
+  })
+
+  describe('userAccountFrom()', () => {
+    describe('in multi user mode', () => {
+      let multiUser = true
+
+      it('should throw an error if no username is passed', () => {
+        let options = { host, multiUser }
+        let accountManager = AccountManager.from(options)
+
+        expect(() => {
+          accountManager.userAccountFrom({})
+        }).to.throw(Error)
+      })
+
+      it('should init webId from param if no username is passed', () => {
+        let options = { host, multiUser }
+        let accountManager = AccountManager.from(options)
+
+        let userData = { webid: 'https://example.com' }
+        let newAccount = accountManager.userAccountFrom(userData)
+        expect(newAccount.webId).to.equal(userData.webid)
+      })
+    })
+
+    describe('in single user mode', () => {
+      let multiUser = false
+
+      it('should not throw an error if no username is passed', () => {
+        let options = { host, multiUser }
+        let accountManager = AccountManager.from(options)
+
+        expect(() => {
+          accountManager.userAccountFrom({})
+        }).to.not.throw(Error)
+      })
+    })
+  })
+
+  describe('addCertKeyToProfile()', () => {
+    let accountManager, certificate, userAccount, profileGraph
+
+    beforeEach(() => {
+      let options = { host }
+      accountManager = AccountManager.from(options)
+      userAccount = accountManager.userAccountFrom({ username: 'alice' })
+      certificate = WebIdTlsCertificate.fromSpkacPost('1234', userAccount, host)
+      profileGraph = {}
+    })
+
+    it('should fetch the profile graph', () => {
+      accountManager.getProfileGraphFor = sinon.stub().returns(Promise.resolve())
+      accountManager.addCertKeyToGraph = sinon.stub()
+      accountManager.saveProfileGraph = sinon.stub()
+
+      return accountManager.addCertKeyToProfile(certificate, userAccount)
+        .then(() => {
+          expect(accountManager.getProfileGraphFor).to
+            .have.been.calledWith(userAccount)
+        })
+    })
+
+    it('should add the cert key to the account graph', () => {
+      accountManager.getProfileGraphFor = sinon.stub()
+        .returns(Promise.resolve(profileGraph))
+      accountManager.addCertKeyToGraph = sinon.stub()
```

Comment:
It's testing that the `addCertKeyToGraph()` is called, and with the specific parameters.